curriculumcourse/src/test/java/curriculumcourse/curriculumcourse/SolverTest.java

```diff
@@ -0,0 +1,67 @@
+package curriculumcourse.curriculumcourse;
+
+import java.util.Arrays;
+
+import org.junit.Test;
+import org.kie.api.KieServices;
+import org.kie.api.runtime.KieContainer;
+import org.optaplanner.core.api.solver.Solver;
+import org.optaplanner.core.api.solver.SolverFactory;
+
+public class SolverTest {
+
+    @Test
+    public void solve() {
+        KieContainer kieContainer = KieServices.Factory.get().getKieClasspathContainer(SolverTest.class.getClassLoader());
+        SolverFactory<CourseSchedule> solverFactory = SolverFactory.createFromKieContainerXmlResource(kieContainer,
+                                                                                                      "curriculumcourse/curriculumcourse/testSolverConfig.xml");
+        Solver<CourseSchedule> solver = solverFactory.buildSolver();
+
+        solver.solve(getSolution());
```

Comment:
If the Solverconfig has no termination or a really long termination, we might want to override it with solverFactory.getSolverConfig().getTerminationConfig().shortenTimeMillisSpentLimit(...). That shortenTimeMillisSpentLimit method is handy.